app/experimenter/experiments/constants/nimbus.py

```diff
@@ -215,6 +215,86 @@ class NimbusTargetingConfig:
     application_choice_names=(Application.DESKTOP.name,),
 )
 
+TARGETING_INFREQUENT_USER_URIS = NimbusTargetingConfig(
+    name="Infrequent user (uris)",
+    slug="infrequent_user_uris",
+    description="Between 1 and 6 URIs visited in the past 28 days",
+    targeting="userMonthlyActivity|length >= 1 && userMonthlyActivity|length <= 6 && "
+    "(currentDate|date - profileAgeCreated|date) / 86400000 >= 28",
+    desktop_telemetry="",
+    application_choice_names=(Application.DESKTOP.name,),
+)
+
+TARGETING_CASUAL_USER_URIS = NimbusTargetingConfig(
+    name="Casual user (uris)",
+    slug="casual_user_uris",
+    description="Between 7 and 13 URIs visited in the past 28 days",
+    targeting="userMonthlyActivity|length >= 7 && userMonthlyActivity|length <= 13 && "
+    "(currentDate|date - profileAgeCreated|date) / 86400000 >= 28",
+    desktop_telemetry="",
+    application_choice_names=(Application.DESKTOP.name,),
+)
+
+TARGETING_REGULAR_USER_URIS = NimbusTargetingConfig(
+    name="Regular user (uris)",
+    slug="regular_user_uris",
+    description="Between 14 and 20 URIs visited in the past 28 days",
+    targeting="userMonthlyActivity|length >= 14 && userMonthlyActivity|length <= 20 && "
+    "(currentDate|date - profileAgeCreated|date) / 86400000 >= 28",
+    desktop_telemetry="",
+    application_choice_names=(Application.DESKTOP.name,),
+)
+
+TARGETING_CORE_USER_URIS = NimbusTargetingConfig(
+    name="Regular user (uris)",
+    slug="regular_user_uris",
+    description="More than 20 URIs visited in the past 28 days",
+    targeting="userMonthlyActivity|length >= 20 && "
```

Comment:
This should be `>= 21` to not overlap with "Regular"

And looks like I misspoke about this being 30 days. A quick check on my main profile `userMonthlyActivity|length` evaluates `Status: ✅, Result: 31` which I suppose makes sense as we include "-30 days" before today, so inclusive of today is 31 days. Let's have a followup on Firefox side to make this 28 days total to be a multiple of a whole week, which avoids some bias if the extra non-week-multiple days happens to fall on a weekend vs weekday.

But these targeting in general should be good as they will work across Fx91 and future versions with the client fix.

Reply:
https://bugzilla.mozilla.org/show_bug.cgi?id=1726739